app/serializers/repository_datatable/repository_number_value_serializer.rb

```diff
@@ -4,7 +4,7 @@ module RepositoryDatatable
   class RepositoryNumberValueSerializer < RepositoryBaseValueSerializer
     def value
       decimals = scope[:column].metadata.fetch('decimals', Constants::REPOSITORY_NUMBER_TYPE_DEFAULT_DECIMALS).to_i
-      value_object.data.round(decimals).to_s('F').remove(/.0+$/)
```

Comment:
isn't 'value_object.data.round(decimals)' just enough?

Reply:
In ruby 3 `.round(0)` will change type to integer and `.to_s('F')` will fail with `no implicit conversion of String into Integer`

Reply:
value_object.data.scale.zero? ? value_object.data : value_object.data.round(decimals)